hq/app/logic/Retry.scala

```diff
@@ -0,0 +1,24 @@
+package logic
+
+import utils.attempt.{Attempt, Failure}
+
+import scala.concurrent.ExecutionContext
+object Retry {
+
+  def until[A](body: => Attempt[A], predicate: A => Boolean, failureMessage: String, maxAttempt: Int = 10)(implicit ec: ExecutionContext): Attempt[A] = {
+
+    def loop(numberOfTries: Int): Attempt[A] = {
+      if (numberOfTries >= maxAttempt ) {
+        Attempt.Left[A](Failure("MAX_ATTEMPT_LIMIT", failureMessage, 500))
```

Comment:
Might be useful to put the friendly message in the log message as well. As it stands we're just going to see MAX_ATTEMPT_LIMIT in the logs where this happens, which isn't heaps of context.

Reply:
how about` s"MAX_ATTEMPT_LIMIT_REACHED: $failureMessage"`